rs/execution_environment/src/history.rs

```diff
@@ -102,6 +105,18 @@ impl IngressHistoryWriterImpl {
             config,
             log,
             received_time: RwLock::new(HashMap::new()),
+            message_state_transition_received_duration_seconds: metrics_registry.histogram(
+                "message_state_transition_received_duration_seconds",
+                "Time taken by messages to get from block maker into ingress queue",
+                // 10ms, 20ms, 50ms, ..., 100s, 200s, 500s
+                decimal_buckets(-2,2),
+            ),
+            message_state_transition_processing_duration_seconds: metrics_registry.histogram(
+                "message_state_transition_processing_duration_seconds",
+                "Time taken by messages to get from block maker to execution",
+                // 10ms, 20ms, 50ms, ..., 100s, 200s, 500s
+                decimal_buckets(-2,2),
+            ),
```

Comment:
I suppose to get induction to execution latency we can simply diff with the above one right? If yes, then I think this is fine. We can construct what we need if we need to. Unless we believe that exporting the metrics directly is substantially better for some reason (e.g. because of some prometheus specific reason when diffing two metrics).

Reply:
Never mind. I also added the third, induction to processing histogram, for the sake of completeness.